src/CommitmentsV2/SFA.DAS.CommitmentsV2/Services/CohortDomainService.cs

```diff
@@ -256,6 +253,19 @@ private static async Task<Provider> GetProvider(long providerId, ProviderCommitm
             return provider;
         }
 
+        private async Task UpdateChangeOfPartyRequest(Cohort cohort, UserInfo userInfo, Party party, CancellationToken cancellationToken)
+        {
+            if (cohort.IsLinkedToChangeOfPartyRequest)
```

Comment:
Instead of doing the update here, I think it will be better to publish the event so the event handler can do the updates.